flang/lib/Optimizer/CodeGen/CodeGen.cpp

```diff
@@ -1311,19 +1312,22 @@ struct EmboxCommonConversion : public FIROpConversion<OP> {
       return rewriter.create<mlir::LLVM::AddressOfOp>(loc, ty,
                                                       global.sym_name());
     }
+    auto i8Ty = rewriter.getIntegerType(8);
+    if (fir::NameUniquer::belongsToModule(
+            name, Fortran::semantics::typeInfoBuiltinModule)) {
+      // Type info derived types do not have type descriptors since they are the
+      // types defining type descriptors.
+      auto i8PtrTy = mlir::LLVM::LLVMPointerType::get(i8Ty);
+      return rewriter.create<mlir::LLVM::NullOp>(loc, i8PtrTy);
```

Comment:
I believe there are helpers someplace in code gen to create the `i8 *` (void ptr) type and a nullptr? (Or maybe they were being removed?)